pkg/volume/flexvolume/flexvolume_test.go

```diff
@@ -123,7 +124,9 @@ func TestCanSupport(t *testing.T) {
 	plugMgr := volume.VolumePluginMgr{}
 	runner := exec.New()
 	installPluginUnderTest(t, "kubernetes.io", "fakeAttacher", tmpDir, execScriptTempl1, nil)
-	plugMgr.InitPlugins(nil, GetDynamicPluginProber(tmpDir, runner), volumetest.NewFakeVolumeHost(t, "fake", nil, nil))
+	if err := plugMgr.InitPlugins(nil, GetDynamicPluginProberWithoutWatcher(tmpDir, runner), volumetest.NewFakeVolumeHost(t, "fake", nil, nil)); err != nil {
```

Comment:
Can you explain how this fixes the tempdir cleanup issue? It's not obvious to me.

Reply:
The` GetDynamicPluginProber() `method starts the real watcher. When `defer os.RemoveAll(tmpDir) `is executed, tmpdir is recreated. I think the real watcher is redundant.